src/main/java/io/github/lambdatest/SmartUIAppSnapshot.java

```diff
@@ -0,0 +1,150 @@
+import com.google.gson.Gson;
+import io.appium.java_client.AppiumDriver;
+import io.github.lambdatest.constants.Constants;
+import io.github.lambdatest.models.*;
+import io.github.lambdatest.utils.GitUtils;
+import io.github.lambdatest.utils.SmartUIUtil;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.OutputType;
+import org.openqa.selenium.TakesScreenshot;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Objects;
+import java.util.logging.Logger;
+
+public class SmartUIAppSnapshot {
+    private final Logger log;
+    private final SmartUIUtil util;
+    private final Gson gson;
+    private String projectToken;
+
+    private BuildData buildData;
+
+    public SmartUIAppSnapshot() {
+        this.log = Logger.getLogger("lt-smart-ui-java-sdk");
+        this.util = new SmartUIUtil();
+        this.gson = new Gson();
+    }
+
+
+    public void start(Map<String, String> options) {
+        if (options == null) {
+            throw new IllegalArgumentException(Constants.Errors.NULL_OPTIONS_OBJECT);
+        }
+        try{
+            this.projectToken = getProjectToken(options);
+            log.info("Project token set as: " + this.projectToken);
+        } catch (Exception e){
+            log.severe(Constants.Errors.PROJECT_TOKEN_UNSET);
+        }
+        Map<String, String> envVars = System.getenv();
+        GitInfo git = GitUtils.getGitInfo(envVars);
+        // Authenticate user and create a build
+        try {
+            BuildResponse buildRes = util.build(git, this.projectToken, options);
+            this.buildData = buildRes.getData();
+            log.info("Build ID set : " + this.buildData.getBuildId() + "for Build name : "+ this.buildData.getName());
+            options.put("buildName", this.buildData.getName());
+        } catch(Exception e) {
+            log.severe("Couldn't create build: " + e.getMessage());
+            throw new IllegalStateException("Couldn't create build: " + e.getMessage());
+        }
+    }
+
+    private String getProjectToken(Map<String, String> options) {
+        if (options != null && options.containsKey("projectToken")) {
+            String token = options.get("projectToken").trim();
+            if (!token.isEmpty()) {
+                return token;
+            }
+        }
+        String envToken = System.getenv("PROJECT_TOKEN");
+        if (envToken != null && !envToken.trim().isEmpty()) {
+            return envToken.trim();
+        }
+        throw new IllegalArgumentException(Constants.Errors.PROJECT_TOKEN_UNSET);
+    }
+
+
+    public void smartUiAppSnapshot(AppiumDriver appiumDriver, String screenshotName, Map<String, String> options) {
+        try {
+            if (Objects.isNull(appiumDriver)) {
+                log.severe(Constants.Errors.SELENIUM_DRIVER_NULL +" during take snapshot");
+                throw new IllegalArgumentException(Constants.Errors.SELENIUM_DRIVER_NULL);
+            }
+            if(Objects.isNull(screenshotName)){
```

Comment:
Here check for  Empty screenshotName is also required, you can use below code
 
 if (snapshotName == null || snapshotName.isEmpty()) {
            log.info(Constants.Errors.SNAPSHOT_NAME_NULL);
            throw new IllegalArgumentException(Constants.Errors.SNAPSHOT_NAME_NULL);
}